core/WASM_core.md

```diff
@@ -0,0 +1,352 @@
+# Summary
+
+This proposal intends to outline the shape of the interfaces and implementation work necessary
+to make Core WASM-compilable.
+
+# Motivation
+
+A few problems a WASM Core Can solve, in no particular order:
+
+* Allow Go and Java to be ported to run on top of Core. *Potentially* without the downsides of CGo
+  and JNI. More on that later.
+* No longer any need to ship compiled Core binaries of various architectures with our existing SDKs.
+  This is both a nice plus for us, in terms of reducing CI complexity, but also a substantial size
+  reduction in our packages.
```

Comment:
> but also a substantial size reduction in our packages

Will be curious what the size of the WASM blob is. I think we may find ourselves wanting to reduce/externalize some third party Rust dependencies.